src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    @Inject
+    private UserService userService;
+
+    public AuthenticationServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> existingUserOptional = userService.findByEmail(email);
+        if (!existingUserOptional.isEmpty()
+                && existingUserOptional.get().getPassword().equals(password)) {

```

Comment:
Instead of using `isEmpty()` and `get()` on the `Optional`, consider using `ifPresent()` or `orElseThrow()` to handle the presence of a user more safely. This will avoid potential `NoSuchElementException` if the `Optional` is empty.